aws/resource_aws_api_gateway_deployment.go

```diff
@@ -172,14 +172,38 @@ func resourceAwsApiGatewayDeploymentDelete(d *schema.ResourceData, meta interfac
 
 	return resource.Retry(5*time.Minute, func() *resource.RetryError {
 		log.Printf("[DEBUG] schema is %#v", d)
-		if _, err := conn.DeleteStage(&apigateway.DeleteStageInput{
+
+		stage, err := conn.GetStage(&apigateway.GetStageInput{
 			StageName: aws.String(d.Get("stage_name").(string)),
 			RestApiId: aws.String(d.Get("rest_api_id").(string)),
-		}); err == nil {
-			return nil
+		})
+		if err != nil {
+			// If the stage is already deleted then there is no need to attempt deletion
+			apigatewayErr, _ := err.(awserr.Error)
+			if apigatewayErr.Code() == "NotFoundException" {
```

Comment:
Nitpick: we have a helper function for this type of logic and prefer the AWS Go SDK constants, e.g. `isAWSErr(err, apigateway.ErrCodeNotFoundException, "")`